scripts/pre_commit_linter.py

```diff
@@ -1762,13 +1639,17 @@ def _check_codeowner_file(verbose_mode_enabled):
 
         # Checks that every file (except those under the dir represented by
         # the dir_patterns) is covered under CODEOWNERS.
+        file_lists = []
```

Comment:
Here and elsewhere -- I don't understand why we're changing the logic, if the aim is just to get this working on Windows. Could you explain?

If we want to do both, I recommend having separate PRs for (a) logic changes, (b) Windows compatibility. That will help us isolate problems in the dev workflow PRs more effectively if any occur.